src/components/SUE/report/SueReportLocation.tsx

```diff
@@ -175,18 +175,13 @@ export const SueReportLocation = ({
     }
   }, []);
 
-  const handleGeocode = async (position: { latitude: number; longitude: number }) => {
-    try {
-      await reverseGeocode({
-        areaServiceData,
-        errorMessage,
-        position,
-        setValue
-      });
-    } catch (error) {
-      throw new Error(error.message);
-    }
-  };
```

Comment:
Since we are not generating errors in this section, the marker in the full screen map view on the report screen is not removed from the screen if it is not at the desired location. This is because the cache section in `SueMapViewScreen` is not working. I prefer to return the error instead of generating an error in the cache section. This way we can add a control according to the return in `SueMapViewScreen`.

|At the moment|should be|
|--|--|
![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-18 at 14 54 00](https://github.com/user-attachments/assets/2ae29698-db9c-44e1-870c-b9e699fbd3b7)|![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-18 at 14 54 06](https://github.com/user-attachments/assets/12f785c6-c30a-4303-ae79-53a19cc1762c)

SueReportLocation:
```
      try {
        await handleGeocode(nativeEvent.coordinate);
      } catch (error) {
        setSelectedPosition(undefined);
        Alert.alert(texts.sue.report.alerts.hint, error.message);
        return { error: error.message };
      }
```

SueMapViewScreen:
```
            const mapPress = await onMapPress({ nativeEvent });

            if (!!mapPress?.error) {
              setSelectedPosition(undefined);
            }
```

Reply:
ok, seems to be neat

Reply:
ok, seems to be neat. can you adjust it please.